### PR TITLE
Listing dashboard Implementation

### PR DESCRIPTION
**Pull request by hugoamaralsantana**
Created the front-end for the Listing Dashboard. It consists of a ListGroup that displays the Listing cards, the Filter Menu, the Full Listing Viewer, and the entire Listing Container that organizes the overall Listing Dashboard layout. 

There are most likely one or two scaling errors with my .scss files where maybe a large window might look slightly different to a small one with the card sizings. But it's a very minor, non-disruptive problem. We can fix it later on.

### CODE REVIEW

File: app/client/components/FilterMenu.jsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            <Dropdown.Item>Experience Decreasing</Dropdown.Item>
+          </DropDownButton>
+          </div>
+          </div>
+      </div>
+    );
+  }
+}
+
+export default withRemote(withSession(withRoutes(withRouter(FilterMenu)))
```
Comment (benmelz) [inline app/client/components/FilterMenu.jsx:45]: I'm guessing this is a copy from other files - you actually don't need these here. They're actually composition hooks that make "global" properties available within the component tree (i.e. `withRemote` exposes the remote instance contained in `App` available as `this.props.remote` here). So, if you don't need to use them within the component the composition isn't necessary.

```suggestion
export default FilterMenu
```

File: app/client/components/FilterMenu.module.scss [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+#sort-container{
+    display: flex;
+    justify-content: space-between;
+    margin-top: 2vh;
+    margin-bottom: 0;
+}
+
+#dropdown-container{
+    margin-right: 1%;
+}
```
Comment (benmelz) [inline app/client/components/FilterMenu.module.scss:55]: As a forewarning, it seems to me that most (if not all) of this css module and the other one can be replaced with bootstrap features - I can go over it with you sometime soon

File: app/client/components/FullListing.jsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  /** @return {JSX.Element} */
+  render() {
+    return (
+        <div id={selectors['full-listing-container']}>
+      <CloseButton id={selectors['close-button']}/>
+      <div id={selectors['listing-container']}>
+        <h1>Software Engineer</h1>
+        <p>This will be the snippet description for each listing. This will be the snippet description for each listing. This will be the snippet description for each listing. This will be the snippet description for each listing.</p>
+      </div>
+      </div>
```
Comment (benmelz) [inline app/client/components/FullListing.jsx:28]: I'm not sure if eslint will pick this up, but indentation here is a little wonky

File: app/client/components/FullListing.module.scss [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    height: 100%;
+    width: auto;
+    border: black;
+    border-radius: 5px;
+    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
+}
+
+#listing-container{
+    margin-top: 20px;
+}
```
Comment (benmelz) [inline app/client/components/FullListing.module.scss:19]: `\n` missing at end of the file (eslint will automatically fix this for you i think)

**Review by benmelz — CHANGES_REQUESTED** [review]
Quick lookthrough - the content itself looks great but a couple of bigger picture things to mention

- From my naked eye it looks to me like there are some linting offenses so if you haven't, `yarn run lint` and fix anything it complains about
- In terms of actual content for the PR you're missing some of the plug-in functionality I was trying (failing) to describe over slack. Specifically I meant stuff like replacing the "sample content" (`FullListing`, `ListingCard`) with proper interpolations from `props` (i.e. replace "This will be the snippet description..." with `{this.props.listing.description}`).

File: app/client/components/FullListing.jsx (hunk 3 above)
Comment (hugoamaralsantana) [reply app/client/components/FullListing.jsx:28]: Yeah, my bad about that. I forgot to run eslint before the push.

File: app/client/components/FilterMenu.module.scss (hunk 2 above)
Comment (hugoamaralsantana) [reply app/client/components/FilterMenu.module.scss:55]: Bet, hmu on Slack

File: app/client/components/FullListing.module.scss (hunk 4 above)
Comment (hugoamaralsantana) [reply app/client/components/FullListing.module.scss:19]: My bad on the eslint stuff. I'll get it touched up as soon as possible.

**Review by benmelz — CHANGES_REQUESTED** [review]
More big-picture comments:

`FilterMenu`:
  - could exist as private component in `ListingBrowser` since definitely only used there
  - dropdown functionality should exist in `ListingBrowser`, thinking modularly a `Menu` should contain only the menu itself and not any wrapping (leave it up to the parent how to display it).
  - can probably omit actual sorting buttons yet (not well defined by team what sorting features we plan to implement ?)

`FullListing`:
  - rename => `Listing`
  - close functionality should exist in `ListingBrowser`, thinking modularly a "Display" component should only contain display functionality for data (leave it up to the parent how to show/hide).
  - use attributes from `this.props.listing` rather than sample text

`ListingCard`:
  - rename => `ListingItem`
  - could exist as private component in `ListingBrowser` since (should be?) only used there
  - use attributes from `this.props.listing` rather than sample text

`ListingContainer`:
  - rename => ListingBrowser
  - copy `this.props.listings` to `this.state.listings` (for filtering/sorting functionality)
  - use `this.state.listings` to generate `ListingItem` (iterative component generation, make sure to set ids, ask if unsure how)

Overall I actually would say to actually hold off on adding responsive features (click listeners etc) - PR is big/complex enough as it is

**Pull request merged**